polar-core/src/error.rs

```diff
@@ -69,7 +68,12 @@ impl PolarError {
                 }
                 _ => {}
             },
-            (e, Some(source), Some(term)) => {
+            (e, Some(source), Some(term))
+            | (
```

Comment:
I think you can drop the arm, and then drop the places where `.set_context(Some(source), None)` are called. It looks to me like the one place that does `set_error_context` on the `kb` should be sufficient.